src/PluginEngine.tsx

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable i18next/no-literal-string */
+import React, { useEffect, Suspense } from "react";
+import { PluginConfigType } from "./Common/hooks/useConfig";
+import { CareAppsContext, useCareApps } from "./Common/hooks/useCareApps";
+import { AppRoutes } from "./Routers/AppRouter";
+import { INavItem } from "./Components/Common/Sidebar/Sidebar";
+import { EnabledPluginConfig, pluginMap } from "./pluginMap";
+import { UserAssignedModel } from "./Components/Users/models";
+import ErrorBoundary from "./Components/Common/ErrorBoundary";
+
+type SupportedPluginExtensions =
+  | "DoctorConnectButtons"
+  | "PatientExternalRegistration";
```

Comment:
Not sure about this, can't we do something like `route` and `id`? 
This is quite restricting, we need to whitelist all locations. 🤔 (Just a thought)